iac-extensions/arm/src/main/java/org/sonar/iac/arm/checks/AnonymousAccessToResourceCheck.java

```diff
@@ -50,12 +56,34 @@ private static void checkWebSites(ContextualResource resource) {
   }
 
   private static void checkWebSitesAuthSettings(ContextualResource contextualResource) {
-    if (!isEqual(AUTH_SETTINGS_V2_RESOURCE_NAME).test(contextualResource.tree.name())) {
+    if (!isEqual(WEBSITES_CONFIG_AUTH_SETTINGS_V2_RESOURCE_NAME).test(contextualResource.tree.name())) {
       return;
     }
 
     ContextualObject globalValidation = contextualResource.object("globalValidation");
     globalValidation.property("requireAuthentication").reportIf(isFalse(), WEBSITES_DISABLED_AUTH_MESSAGE);
     globalValidation.property("unauthenticatedClientAction").reportIf(isEqual("AllowAnonymous"), WEBSITES_DISABLED_AUTH_MESSAGE);
   }
+
+  private static void checkApiManagementService(ContextualResource resource) {
+    ContextualResource signIn = resource.childResourceBy("portalsettings", it -> isEqual(APIMGMT_PORTALSETTINGS_SIGNIN_RESOURCE_NAME).test(it.name()));
+
+    if (signIn.isAbsent()) {
+      resource.report(APIMGMT_MISSING_SIGN_IN_RESOURCE_MESSAGE);
+    } else {
+      checkApiManagementPortalSettings(signIn);
+    }
+  }
+
+  private static void checkApiManagementPortalSettings(ContextualResource resource) {
+    if (!isEqual(APIMGMT_PORTALSETTINGS_SIGNIN_RESOURCE_NAME).test(resource.tree.name())) {
+      return;
+    }
+
+    resource.property("enabled").reportIf(isFalse(), APIMGMT_PORTAL_SETTINGS_DISABLED_MESSAGE);
+
+    resource.childResourceBy("apis", r -> true)
+      .property("authenticationSettings")
+      .reportIfAbsent(APIMGMT_AUTHENTICATION_SETTINGS_NOT_SET_MESSAGE);
+  }
```

Comment:
Consider move this logic to separate method. IMO it can produce FP for `Microsoft.ApiManagement/service/portalsettings/signIn/apis` (I'm not sure if such resource is valid).
Also the logic in code should be similar as described in ticket. Now `checkApiManagementPortalSettings` is called for 2 different types, where the ticket descriptions mention `authenticationSettings` only once.

Reply:
Good catch, I placed this check in an incorrect place. Moved and added more test cases.

> Now `checkApiManagementPortalSettings` is called for 2 different types

It's actually called for the same type, but the type can be either standalone or nested. I believe our `AbstractArmResourceCheck#register` doesn't handle nested types